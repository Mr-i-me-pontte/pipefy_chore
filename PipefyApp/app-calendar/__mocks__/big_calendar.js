export const label = 'Testing label';
export const onNavigate = jest.fn();
export const onViewChange = jest.fn();
export const view = 'month';
export const views = ['month', 'week', 'day', 'agenda'];

export default { label, onNavigate, onViewChange, view, views };
