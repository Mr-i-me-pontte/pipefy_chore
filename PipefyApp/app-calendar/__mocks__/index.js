import MockDate from 'mockdate';

MockDate.set('2017-12-01');

export const card = {
  due_date: '2017-12-15T16:00:00-02:00',
  id: 'aBc_1234',
  internalId: '1',
  isVisible: true,
  title: 'Testing card',
};

export const e = { preventDefault: jest.fn() };

export const event = {
  end: new Date(new Date(card.due_date).getTime() + 30 * 60000),
  id: card.id,
  start: new Date(card.due_date),
  title: card.title,
};

export const pipefy = {
  app: { pipeId: '1' },
  locale: 'en',
  openCard: jest.fn(),
  allCards: () => Promise.resolve({ cards: [card], pageInfo: { hasNextPage: false } })
};
