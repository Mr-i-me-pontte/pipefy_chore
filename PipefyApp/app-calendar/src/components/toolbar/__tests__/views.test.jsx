import React from 'react';
import renderer from 'react-test-renderer';

import Views from '../views';
import { onViewChange, views } from '../../../../__mocks__/big_calendar';

describe('Views', () => {
  const component = renderer.create(
    <Views availableViews={views} currentView="month" handleViewChange={onViewChange} />
  );

  it('render', () => {
    expect(component.toJSON()).toMatchSnapshot();
    expect(onViewChange).not.toBeCalled();
  });

  it('calls handle view change', () => {
    component.toJSON().children[1].props.onClick();
    expect(onViewChange).toBeCalledWith(views[1]);
  });
});
