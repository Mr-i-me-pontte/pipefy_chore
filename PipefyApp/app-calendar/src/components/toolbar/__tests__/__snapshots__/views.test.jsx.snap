// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views render 1`] = `
<span
  className="rbc-btn-group"
>
  <button
    className="rbc-active"
    onClick={[Function]}
    tabIndex="0"
  >
    month
  </button>
  <button
    className=""
    onClick={[Function]}
    tabIndex="0"
  >
    week
  </button>
  <button
    className=""
    onClick={[Function]}
    tabIndex="0"
  >
    day
  </button>
  <button
    className=""
    onClick={[Function]}
    tabIndex="0"
  >
    agenda
  </button>
</span>
`;
